# Checking dirs that had error

## 1

In [1]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 4/3_92_1_1335/3_92_1_1335_5/Audible"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 226, in main
    extract_metadata_of_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 164, in extract_metadata_of_files
    tuple_date, tuple_metadata_of_file = extract_metadata(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 28, in extract_metadata
    tuple_metadata_of_file = (filename,read_metadata_audio.get_metadata_of_file(filename))
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 121, in get_metadata_of_file
    date_with_timezone = get_date_with_timezone(comment_metadata)
  File 

CalledProcessError: Command 'b'dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 4/3_92_1_1335/3_92_1_1335_5/Audible"\nextract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"\n'' returned non-zero exit status 1.

Problematic file:

/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 4/3_92_1_1335/3_92_1_1335_5/Audible/19700101_000006.WAV

In [2]:
%%bash
filename="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 4/3_92_1_1335/3_92_1_1335_5/Audible/19700101_000006.WAV"
exiftool -a -j "$filename"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 4/3_92_1_1335/3_92_1_1335_5/Audible/19700101_000006.WAV",
  "ExifToolVersion": 12.40,
  "FileName": "19700101_000006.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 4/3_92_1_1335/3_92_1_1335_5/Audible",
  "FileSize": "488 bytes",
  "FileModifyDate": "1961:11:25 17:31:50-06:00",
  "FileAccessDate": "2022:03:10 08:27:02-06:00",
  "FileInodeChangeDate": "2022:02:11 17:07:28-06:00",
  "FilePermissions": "-rwxrwxrwx",
  "FileType": "WAV",
  "FileTypeExtension": "wav",
  "MIMEType": "audio/x-wav",
  "Encoding": "Microsoft PCM",
  "NumChannels": 1,
  "SampleRate": 48000,
  "AvgBytesPerSec": 96000,
  "BitsPerSample": 16,
  "Comment": "Recorded at 00:00:06 01/01/1970 (UTC) by AudioMoth 248D9B026037BB7D at medium gain setting while battery state was 4.7V and temperature was 30.2C. Recording cancelled before completion due to change of switch position.",
  "Artist": "AudioMoth 248D9B026037BB7D",
  "Duration": "0.01 s"
}]


File was only of .01 segs and is the only file in dir. Moving dir to dir

Moving dir to directory "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 4 error/3_92_1_1335/3_92_1_1335_5/"

In [3]:
%%bash
mv "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 4/3_92_1_1335/3_92_1_1335_5/Audible/" "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 4 error/3_92_1_1335/3_92_1_1335_5/"

Manually removing dir /LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 4/3_92_1_1335/3_92_1_1335_5/Audible from /shared_volume/sipecam_subdirectories_09-03-2022.txt

# 2

In [9]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 226, in main
    extract_metadata_of_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 164, in extract_metadata_of_files
    tuple_date, tuple_metadata_of_file = extract_metadata(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 27, in extract_metadata
    date_of_file = read_metadata_audio.extract_date(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 143, in extract_date
    return get_date(comment_metadata)#example: 2021-07-28
  File "/usr/local/lib/python3.8/dist-packages/simex

CalledProcessError: Command 'b'dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible"\nextract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"\n'' returned non-zero exit status 1.

Problematic file:

In [6]:
%%bash
filename="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible/19700110_173320.WAV"
exiftool -a -j "$filename"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible/19700110_173320.WAV",
  "ExifToolVersion": 12.40,
  "FileName": "19700110_173320.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible",
  "FileSize": "0 bytes",
  "FileModifyDate": "2021:09:26 05:30:18-05:00",
  "FileAccessDate": "2022:03:10 08:38:40-06:00",
  "FileInodeChangeDate": "2022:01:18 15:34:36-06:00",
  "FilePermissions": "-rwxrwxrwx",
  "Error": "File is empty"
}]


CalledProcessError: Command 'b'filename="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible/19700110_173320.WAV"\nexiftool -a -j "$filename"\n'' returned non-zero exit status 1.

In [10]:
%%bash
filename="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible/19700110_172413.WAV"
exiftool -a -j "$filename"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible/19700110_172413.WAV",
  "ExifToolVersion": 12.40,
  "FileName": "19700110_172413.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible",
  "FileSize": "0 bytes",
  "FileModifyDate": "2021:09:26 05:30:18-05:00",
  "FileAccessDate": "2022:03:10 08:52:31-06:00",
  "FileInodeChangeDate": "2022:01:18 15:34:36-06:00",
  "FilePermissions": "-rwxrwxrwx",
  "Error": "File is empty"
}]


CalledProcessError: Command 'b'filename="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible/19700110_172413.WAV"\nexiftool -a -j "$filename"\n'' returned non-zero exit status 1.

Moving files to directory "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2 error/3_92_0_1342/1342_1/audible/"

In [7]:
%%bash
mv "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible/19700110_173320.WAV" "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2 error/3_92_0_1342/1342_1/audible/"

In [11]:
%%bash
mv "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible/19700110_172413.WAV" "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2 error/3_92_0_1342/1342_1/audible/"

Implemented handling of empty audio files in: [commit](https://github.com/CONABIO/sipecam-metadata-extractor/blob/262f0ee38f748ca283e4bc87720adc07e8cb6dd4/src/simex/check_if_there_are_audio_empty_files_in_dir_and_move_them.py)

Rerunning cmd

In [1]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible"
check_if_there_are_audio_empty_files_in_dir_and_move_them --input_directory_audio "$dir_to_be_processed" --number_of_processes 10

In [4]:
%%bash
ls "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible/error"

19700110_173308.WAV
19700110_173332.WAV
19700110_173349.WAV
19700110_173937.WAV
19700110_174020.WAV
19700110_174026.WAV
19700110_174032.WAV


In [2]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_0_1342/1342_1/audible"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed" --parallel --number_of_processes 10

# 3

dir: /LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_1_1340/1340_2/audible

Checking if there is empty content in dir for files

In [3]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_1_1340/1340_2/audible"
check_if_there_are_audio_empty_files_in_dir_and_move_them --input_directory_audio "$dir_to_be_processed" --number_of_processes 10

In [5]:
%%bash
ls "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_1_1340/1340_2/audible/error"

20211003_202000.WAV


In [4]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 2/3_92_1_1340/1340_2/audible"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed" --parallel --number_of_processes 10

# 4

dir: /LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783

In [5]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783"
check_if_there_are_audio_empty_files_in_dir_and_move_them --input_directory_audio "$dir_to_be_processed" --number_of_processes 10

In [7]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 221, in main
    extract_metadata_of_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 159, in extract_metadata_of_files
    tuple_date, tuple_metadata_of_file = extract_metadata(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 28, in extract_metadata
    tuple_metadata_of_file = (filename,read_metadata_audio.get_metadata_of_file(filename))
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 121, in get_metadata_of_file
    date_with_timezone = get_date_with_timezone(comment_metadata)
  File 

CalledProcessError: Command 'b'dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783"\nextract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"\n'' returned non-zero exit status 1.

Problematic file:

In [8]:
%%bash
filename="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/19700101_002058.WAV"
exiftool -a -j "$filename"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/19700101_002058.WAV",
  "ExifToolVersion": 12.40,
  "FileName": "19700101_002058.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783",
  "FileSize": "224 KiB",
  "FileModifyDate": "1961:11:25 17:52:44-06:00",
  "FileAccessDate": "2022:03:10 12:31:46-06:00",
  "FileInodeChangeDate": "2022:02:11 16:41:59-06:00",
  "FilePermissions": "-rwxrwxrwx",
  "FileType": "WAV",
  "FileTypeExtension": "wav",
  "MIMEType": "audio/x-wav",
  "Encoding": "Microsoft PCM",
  "NumChannels": 1,
  "SampleRate": 48000,
  "AvgBytesPerSec": 96000,
  "BitsPerSample": 16,
  "Comment": "Recorded at 00:20:58 01/01/1970 (UTC) by AudioMoth 24E1440360371EA1 at medium gain setting while battery state was 4.8V and temperature was 32.9C. Recording cancelled before completion due to change of switch position.",
  "Artist": "AudioMoth 24E1440360371EA1",
  "Duration": "2.39 s"
}]


File was only of 2.39 segs

Moving file to directory "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/error"

In [9]:
%%bash
mv "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/19700101_002058.WAV" "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/error/"

Rerunning cmd:

In [11]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 221, in main
    extract_metadata_of_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 159, in extract_metadata_of_files
    tuple_date, tuple_metadata_of_file = extract_metadata(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 28, in extract_metadata
    tuple_metadata_of_file = (filename,read_metadata_audio.get_metadata_of_file(filename))
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 121, in get_metadata_of_file
    date_with_timezone = get_date_with_timezone(comment_metadata)
  File 

CalledProcessError: Command 'b'dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783"\nextract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"\n'' returned non-zero exit status 1.

In [12]:
%%bash
filename="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/19700101_001629.WAV"
exiftool -a -j "$filename"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/19700101_001629.WAV",
  "ExifToolVersion": 12.40,
  "FileName": "19700101_001629.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783",
  "FileSize": "488 bytes",
  "FileModifyDate": "1961:11:25 17:48:14-06:00",
  "FileAccessDate": "2022:03:10 14:18:29-06:00",
  "FileInodeChangeDate": "2022:02:11 16:41:59-06:00",
  "FilePermissions": "-rwxrwxrwx",
  "FileType": "WAV",
  "FileTypeExtension": "wav",
  "MIMEType": "audio/x-wav",
  "Encoding": "Microsoft PCM",
  "NumChannels": 1,
  "SampleRate": 48000,
  "AvgBytesPerSec": 96000,
  "BitsPerSample": 16,
  "Comment": "Recorded at 00:16:29 01/01/1970 (UTC) by AudioMoth 24E1440360371EA1 at medium gain setting while battery state was 4.7V and temperature was 32.3C. Recording cancelled before completion due to change of switch position.",
  "Artist": "AudioMoth 24E1440360371EA1",
  "Duration": "0.01 s"
}]


File was only of .01 segs

Moving file to directory "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/error"

In [13]:
%%bash
mv "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/19700101_001629.WAV" "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/error/"

Rerunning cmd:

In [15]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 221, in main
    extract_metadata_of_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 159, in extract_metadata_of_files
    tuple_date, tuple_metadata_of_file = extract_metadata(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 28, in extract_metadata
    tuple_metadata_of_file = (filename,read_metadata_audio.get_metadata_of_file(filename))
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 121, in get_metadata_of_file
    date_with_timezone = get_date_with_timezone(comment_metadata)
  File 

CalledProcessError: Command 'b'dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783"\nextract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"\n'' returned non-zero exit status 1.

In [16]:
%%bash
filename="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/19700101_001632.WAV"
exiftool -a -j "$filename"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/19700101_001632.WAV",
  "ExifToolVersion": 12.40,
  "FileName": "19700101_001632.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783",
  "FileSize": "24 MiB",
  "FileModifyDate": "1961:11:25 17:52:40-06:00",
  "FileAccessDate": "2022:03:10 14:39:24-06:00",
  "FileInodeChangeDate": "2022:02:11 16:41:59-06:00",
  "FilePermissions": "-rwxrwxrwx",
  "FileType": "WAV",
  "FileTypeExtension": "wav",
  "MIMEType": "audio/x-wav",
  "Encoding": "Microsoft PCM",
  "NumChannels": 1,
  "SampleRate": 48000,
  "AvgBytesPerSec": 96000,
  "BitsPerSample": 16,
  "Comment": "Recorded at 00:16:32 01/01/1970 (UTC) by AudioMoth 24E1440360371EA1 at medium gain setting while battery state was 4.8V and temperature was 32.5C. Recording cancelled before completion due to change of switch position.",
  "Artist": "AudioMoth 24E1440360371EA1",
  "Duration": "0:04:25"
}]


Observe date is wrong... 1970

As there were several files with "cancelled" word then [commit](https://github.com/CONABIO/sipecam-metadata-extractor/commit/03133fb00a91535d2a806bb445df1d01ac8e9793) was implemented in `check_if_there_are_audio_empty_files_in_dir_and_move_them` cli 

In [1]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/"
check_if_there_are_audio_empty_files_in_dir_and_move_them --input_directory_audio "$dir_to_be_processed" --number_of_processes 10

In [2]:
%%bash
ls "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/error"

19700101_001629.WAV
19700101_001632.WAV
19700101_002058.WAV


In [3]:
%%bash
exiftool -j -a "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/error/19700101_001629.WAV"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/error/19700101_001629.WAV",
  "ExifToolVersion": 12.40,
  "FileName": "19700101_001629.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783/error",
  "FileSize": "488 bytes",
  "FileModifyDate": "1961:11:25 17:48:14-06:00",
  "FileAccessDate": "2022:03:10 14:27:18-06:00",
  "FileInodeChangeDate": "2022:03:10 14:28:34-06:00",
  "FilePermissions": "-rwxrwxrwx",
  "FileType": "WAV",
  "FileTypeExtension": "wav",
  "MIMEType": "audio/x-wav",
  "Encoding": "Microsoft PCM",
  "NumChannels": 1,
  "SampleRate": 48000,
  "AvgBytesPerSec": 96000,
  "BitsPerSample": 16,
  "Comment": "Recorded at 00:16:29 01/01/1970 (UTC) by AudioMoth 24E1440360371EA1 at medium gain setting while battery state was 4.7V and temperature was 32.3C. Recording cancelled before completion due to change of switch position.",
  "Artist": "AudioMoth 24E1440360371EA1",
  "Duration": "0.01 s"
}]


Rerunning cmd:

In [6]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_0_1343/Ch1/10306783"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed" --parallel --number_of_processes 10

# 5

In [7]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_1_1334/09048594/10306516 ultra"
check_if_there_are_audio_empty_files_in_dir_and_move_them --input_directory_audio "$dir_to_be_processed" --number_of_processes 10

In [8]:
%%bash
ls "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_1_1334/09048594/10306516 ultra/error"

19700101_000000.WAV
20211021_044000.WAV


Rerunning cmd:

In [11]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 3/3_92_1_1334/09048594/10306516 ultra/"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed" --parallel --number_of_processes 10

# 6

In [12]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 1/3_92_1_1338/Audio/1338_1/Audible"
check_if_there_are_audio_empty_files_in_dir_and_move_them --input_directory_audio "$dir_to_be_processed" --number_of_processes 10

In [14]:
#%%bash
#ls "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 1/3_92_1_1338/Audio/1338_1/Audible/error"

Rerunning cmd:

In [15]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 1/3_92_1_1338/Audio/1338_1/Audible"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 221, in main
    extract_metadata_of_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 159, in extract_metadata_of_files
    tuple_date, tuple_metadata_of_file = extract_metadata(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 27, in extract_metadata
    date_of_file = read_metadata_audio.extract_date(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 143, in extract_date
    return get_date(comment_metadata)#example: 2021-07-28
  File "/usr/local/lib/python3.8/dist-packages/simex

CalledProcessError: Command 'b'dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 1/3_92_1_1338/Audio/1338_1/Audible"\nextract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"\n'' returned non-zero exit status 1.

In [16]:
%%bash
filename="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 1/3_92_1_1338/Audio/1338_1/Audible/20210809_035000.WAV"
exiftool -j -a "$filename"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 1/3_92_1_1338/Audio/1338_1/Audible/20210809_035000.WAV",
  "ExifToolVersion": 12.40,
  "FileName": "20210809_035000.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 1/3_92_1_1338/Audio/1338_1/Audible",
  "FileSize": "5.5 MiB",
  "FileModifyDate": "2021:10:27 12:45:24-05:00",
  "FileAccessDate": "2022:03:11 04:53:10-06:00",
  "FileInodeChangeDate": "2022:02:11 16:05:42-06:00",
  "FilePermissions": "-rwxrwxrwx",
  "Error": "Entire file is binary zeros"
}]


CalledProcessError: Command 'b'filename="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 1/3_92_1_1338/Audio/1338_1/Audible/20210809_035000.WAV"\nexiftool -j -a "$filename"\n'' returned non-zero exit status 1.

[commit](https://github.com/CONABIO/sipecam-metadata-extractor/commit/bf171b1b1ab08e93f444e4878422f5c0056f5574) was implemented in `check_if_there_are_audio_empty_files_in_dir_and_move_them` cli to check for words "binary zeros"

In [12]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 1/3_92_1_1338/Audio/1338_1/Audible"
check_if_there_are_audio_empty_files_in_dir_and_move_them --input_directory_audio "$dir_to_be_processed" --number_of_processes 10

In [14]:
%%bash
ls "/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 1/3_92_1_1338/Audio/1338_1/Audible/error"

Rerunning cmd:

In [28]:
#%%bash
#dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_92/Mes 1/3_92_1_1338/Audio/1338_1/Audible"
#extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"

TODO: include in mv cli name of date of execution of mv cli for json file

TODO: include in extract cli handling of GPS with try... except for dirs with images that have no GPS key in metadata